test/e2e/privileged.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	api "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/latest"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+)
+
+const (
+	privilegedPodName       = "privileged-pod"
+	privilegedContainerName = "privileged-container"
+	imageName               = "ubuntu:14.04"
+	privilegedCommand       = "ip link add dummy1 type dummy"
+)
+
+type PrivilegedPodTestConfig struct {
+	privilegedPod *api.Pod
+	f             *Framework
+}
+
+var _ = Describe("PrivilegedPod", func() {
+	f := NewFramework("e2e-privilegedpod")
+	config := &PrivilegedPodTestConfig{
+		f: f,
+	}
+	It("should test privileged pod", func() {
+		By("Creating a privileged pod")
+		config.createPrivilegedPod()
+
+		By("Executing privileged command on privileged pod")
+		config.runPrivilegedCommand()
+	})
+})
+
+func (config *PrivilegedPodTestConfig) runPrivilegedCommand() {
+	stdout, stderr, err := config.f.kubectlExec(config.f.Namespace.Name, privilegedPodName, privilegedContainerName, "--", "/bin/sh", "-c", privilegedCommand)
+	Logf("command:%s, stdout:%s, stderr:%s, err:%v", privilegedCommand, string(stdout), string(stderr), err)
```

Comment:
when printing strings use %q instead of %s so that they are automatically quoted.